lib/dag/edges.rb

```diff
@@ -1,59 +1,69 @@
 module Dag
   module Edges
-
     def self.included(base)
-      base.send :include, EdgeInstanceMethods
+      base.send(:include, EdgeInstanceMethods)
+    end
+
+    def ancestor_scoped_record_id(ancestor)
+      if ancestor.respond_to?(:scoped_record_id)
+        ancestor.scoped_record_id
+      elsif scoped_record_id_column_name
+        ancestor.public_send(scoped_record_id_column_name)
+      end
     end
 
-    #Class methods that extend the link model for both polymorphic and non-polymorphic graphs
-    #Returns a new edge between two points
+    # Class methods that extend the link model for both polymorphic and non-polymorphic graphs
+    # Returns a new edge between two points
     def build_edge(ancestor, descendant)
-      source = self::EndPoint.from(ancestor)
-      sink = self::EndPoint.from(descendant)
-      conditions = self.conditions_for(source, sink)
-      path = self.new(conditions)
+      scoped_record_id = ancestor_scoped_record_id(ancestor)
```

Comment:
Původně jsem chtěl volat `def build_edge(ancestor, descendant, scoped_record_id)`, ale jak se to pořád zanořuje dál, tak bylo jednodušší brát to id z `ancestor`.